### PR TITLE
test: v2 sample test supports multiple test cases & internal endpoint

### PR DESCRIPTION
**Pull request by Bobgy**
**Description of your changes:**
Changes:
* v2 sample test supports multiple test cases -- many samples need more than one set of tests, either testing both v1 and v2 compatible versions, or testing multiple set of arguments, as shown in https://github.com/kubeflow/pipelines/pull/5431
* use internal endpoint for accessing KFP API server -- so that when errors happen, we can see the detailed error log (we preserve URLs as external links by passing in an extra parameter called external_host)

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?

    [Learn more about cherry-picking updates into the release branch](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#cherry-picking-pull-requests-to-release-branch).

### CODE REVIEW

File: samples/test/util.py [hunk 1]
```diff
@@ -1,55 +1,104 @@
...
+
+    _run_test(test_wrapper)
+
+
+def _run_test(callback):
+
     def main(
-        output_directory: str,  # example gs://your-bucket/path/to/workdir
-        host: str = 'http://ml-pipeline:8888',
+        output_directory: str = None,  # example
```
Comment (neuromage) [inline samples/test/util.py:44]: nit: type these as Optional[str] since None is a valid value.

File: samples/test/util.py [hunk 2]
```diff
@@ -1,55 +1,104 @@
...
-        print("Run details page URL:")
-        print(f"{host}/#/runs/details/{run_result.run_id}")
-        run_response = run_result.wait_for_run_completion(10 * MINUTE)
-        run = run_response.run
-        from pprint import pprint
-        pprint(run_response.run)
-        print("Run details page URL:")
-        print(f"{host}/#/runs/details/{run_result.run_id}")
-        verify_func(run_id=run_result.run_id, run=run)
+            mode: str = kfp.dsl.PipelineExecutionMode.V2_COMPATIBLE,
```
Comment (neuromage) [inline samples/test/util.py:78]: nit: the type is actually the enum class rather than str

File: samples/test/util.py (hunk 1 above)
Comment (Bobgy) [reply samples/test/util.py:44]: Thank you! Now I know python typings better

**Comment by Bobgy** [comment]
Thank you @neuromage!
I fixed the issues, would you want to confirm again?

**Pull request merged**